pkg/action/executor/machine_scalers.go

```diff
@@ -53,69 +54,30 @@ func (client *k8sClusterApi) verifyClusterAPIEnabled() error {
 	return nil
 }
 
-// identifyManagingMachine returns the Machine that manages a Machine.
-// An error is returned if the Machine is not found.
-func (client *k8sClusterApi) identifyManagingMachine(machineName string) (*clusterv1.Machine, error) {
-	machine, err := client.machine.Get(machineName, metav1.GetOptions{})
-	if err == nil {
-		return machine, nil
-	}
-	nodes := client.k8sClient.CoreV1().Nodes()
-	node, err := nodes.Get(machineName, metav1.GetOptions{})
+// identifyManagingMachine returns the Machine that manages the given node.
+// An error is returned if the Machine is not found or the node does not exist.
+func (client *k8sClusterApi) identifyManagingMachine(nodeName string) (*machinev1beta1.Machine, error) {
+	// This step simply verifies that the node exists
+	_, err := client.k8sClient.CoreV1().Nodes().Get(nodeName, metav1.GetOptions{})
 	if err != nil {
-		return nil, err
-	}
-	nodeSpecTmp := strings.Split(node.Spec.ProviderID, "/")
-	if len(nodeSpecTmp) < 2 {
-		return nil, fmt.Errorf("Node " + machineName + " has no valid provider ID")
+		return nil, fmt.Errorf("Error retrieving node %s: %v", nodeName, err)
 	}
-	nodeProviderID := nodeSpecTmp[len(nodeSpecTmp)-1]
-	// List all machines and match.
+
+	// List all machines and match the node.
 	machineList, err := client.machine.List(metav1.ListOptions{})
 	if err != nil {
 		return nil, err
 	}
 	for _, machine := range machineList.Items {
-		machineSpecTmp := strings.Split(*machine.Spec.ProviderID, "/")
-		if len(nodeSpecTmp) < 2 {
-			return nil, fmt.Errorf("Machine " + machine.Name + " has no valid provider ID")
-		}
-		machineProviderID := machineSpecTmp[len(machineSpecTmp)-1]
-		if machineProviderID == nodeProviderID {
+		if nodeName == machine.Status.NodeRef.Name {
```

Comment:
I remembered when Michael was implementing the AWS version, the provider ID is the only attr that matches. Can we make it match both on Name or ProviderID.

Reply:
@enlinxu I don't think its a good idea. I think the `NodeRef` is an authoritative link in terms of the API. Using the providerID was anyways a hacky mechanism earlier.